### PR TITLE
Remove unused FPE initialization from dot.c

### PR DESCRIPTION
**Pull request by ErwinJanssen**
In cmd/dot/dot.c, non standard floating point exceptions are enabled in a platform specific way in a sequence of `ifdef`s. However, this is never used because:
- The rest of the code never checks for set floating point exception flags (with `fetestexcept` for example).
- The check that toggles the whole block, `ifndef NO_FPERR` is almost false. `NO_FPERR` is always defined on Windows and only disabled on specific platforms that do not support `-ffast-math`.

In conclusion, the code adds unnecessary complexity and should be removed. Do you agree or am I missing something?

### CODE REVIEW

**Comment by ellson** [comment]
My recollection is that this was added so that we would catch some FP exceptions.   I think that the code is arithmetically stable these days such that we don't generate such exceptions,  so maybe removing will not cause any change in behavior,  but  I'm a little concerned that we might be removing a facility that will aid in catching problems in the future?

I think the reason that there is not code to handle the exception is because we wanted to catch the uncaught exception signals as a crash, so that they could be debugged.

I see change 8d333ecf806633b9284d9c313a30faf74a73e221 where the exceptions were added apparently after some problems with cairo were resolved.

I see change 2ea1340c3d318d08e48dab3602da00dee9eaaca5 where -DNO_FPERR was moved to config.h
such that if -ffast-math is available then NO_FPERR is set.

Perhaps there are no platforms without -ffast-math these days?     What about Raspberry-PIs ?  Centos4, ...???

I can see your point that this code is probably not used on the majority of platforms today.   I'm just a little reluctant to remove because of:
     a) the more obscure platforms
     b) the ability to catch errors introduced in the future.

I don't feel really strongly about this.   Perhaps Stephen and Emden can offer their opions

**Comment by ErwinJanssen** [comment]
I agree, adding future floating point exceptions should not be increasingly difficult and more "obscure" platforms should still be supported.

But I think the removed code isn't required for testing floating point exceptions. At the page http://en.cppreference.com/w/c/numeric/fenv/FE_exceptions is the following example program:

```
#include <stdio.h>
#include <math.h>
#include <float.h>
#include <fenv.h>
 
#pragma STDC FENV_ACCESS ON
void show_fe_exceptions(void)
{
    printf("exceptions raised:");
    if(fetestexcept(FE_DIVBYZERO)) printf(" FE_DIVBYZERO");
    if(fetestexcept(FE_INEXACT))   printf(" FE_INEXACT");
    if(fetestexcept(FE_INVALID))   printf(" FE_INVALID");
    if(fetestexcept(FE_OVERFLOW))  printf(" FE_OVERFLOW");
    if(fetestexcept(FE_UNDERFLOW)) printf(" FE_UNDERFLOW");
    feclearexcept(FE_ALL_EXCEPT);
    printf("\n");
}
 
int main(void)
{
    printf("MATH_ERREXCEPT is %s\n",
           math_errhandling & MATH_ERREXCEPT ? "set" : "not set");
 
    printf("0.0/0.0 = %f\n", 0.0/0.0);
    show_fe_exceptions();
 
    printf("1.0/0.0 = %f\n", 1.0/0.0);
    show_fe_exceptions();
 
    printf("1.0/10.0 = %f\n", 1.0/10.0);
    show_fe_exceptions();
 
    printf("sqrt(-1) = %f\n", sqrt(-1));
    show_fe_exceptions();
 
    printf("DBL_MAX*2.0 = %f\n", DBL_MAX*2.0);
    show_fe_exceptions();
 
    printf("nextafter(DBL_MIN/pow(2.0,52),0.0) = %.1f\n",
                      nextafter(DBL_MIN/pow(2.0,52),0.0));
    show_fe_exceptions();
}
```

When executed, it prints the raised exceptions. No `feenableexcept` or similar is required.

**Comment by ellson** [comment]
Fair enough.
